### ISSUE TITLE
Page template: template previews showing old global styles font

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#40353]
#### Steps to reproduce

1. Modify your homepage as homepage template
1. Adjust global styles font from the sidebar. Save everything, remember to press "publish"  from global styles sidebar. At this point feel free to refresh the browser as the font mismatches might get more obvious only then.
    <img width="1296" alt="Screenshot 2020-03-21 at 23 12 59" src="https://user-images.githubusercontent.com/87168/77236732-e4d53d00-6bc9-11ea-99c3-99329ec9a0f8.png">
1. Now if you press "change layout", you'll get the preview with previous global styles font:
    <img width="1440" alt="Screenshot 2020-03-21 at 23 13 09" src="https://user-images.githubusercontent.com/87168/77236738-f0c0ff00-6bc9-11ea-8320-0696689d475f.png">
1. Small preview in the sidebar is mismatching too as you do more modifications to global styles font:
    <img width="1440" alt="Screenshot 2020-03-21 at 23 13 52" src="https://user-images.githubusercontent.com/87168/77236756-2665e800-6bca-11ea-812e-93bb88a5b80f.png">

#### What I expected
- My active font show up in previews each time

#### What happened instead
- Old fonts visible

#### Browser / OS version
Latest Chrome.

cc @Automattic/ajax

**Comment by frontdevde** [comment]
I can reliably reproduce the issue:
Start with a new page and select a page layout. In the editor then change to a different global style and see the page layout update in the editor as expected. Now click "Change layout" and in the template selector preview the same layout, it doesn't show the update global style. Selecting the layout again and then changing the layout again doesn't change this. Only after a hard refresh, the new global styles are applied in the preview.

> Small preview in the sidebar is mismatching too as you do more modifications to global styles font

In the current iteration of the feature, this behavior is expected as the small preview in the sidebar utilizes the static screenshots that are also used for the small previews in the layout selector. I'd suggest logging this as a feature request in a separate issue.

**Comment by frontdevde** [comment]
I think I might have tracked down the source of the issue:

When a new Global Style is selected, the Global styles are applied inline to the `.editor-styles-wrapper` element thus overwriting the previous configuration.

<img width="1475" alt="Screenshot 2020-03-25 at 16 42 31" src="https://user-images.githubusercontent.com/1562646/77556204-7a1d4d80-6eb8-11ea-90cb-b078f5869b07.png">

In the instance of the `.editor-styles-wrapper` inside the SPT iframe preview, these styles are then missing:

<img width="1478" alt="Screenshot 2020-03-25 at 16 43 51" src="https://user-images.githubusercontent.com/1562646/77556520-e8faa680-6eb8-11ea-8226-12d511dd86d7.png">

Adding those styles inline yields the desired result:

<img width="1476" alt="Screenshot 2020-03-25 at 16 44 54" src="https://user-images.githubusercontent.com/1562646/77556554-f879ef80-6eb8-11ea-84a6-7bf4c899867a.png">

This is also the reason why it works after a hard refresh because then the new styles have become the default and the inline styles are no longer needed.

To fix this we will have to make sure that the inline styles we see on `.editor-styles-wrapper` are applied to the instance in the preview iframe.

**Comment by frontdevde** [comment]
Side-note: it looks like previews are generally not reacting to Global Styles changes at the moment.

Default after reload:

<img width="1353" alt="Screenshot 2020-03-25 at 18 18 20" src="https://user-images.githubusercontent.com/1562646/77566566-20238480-6ec6-11ea-997c-872980bbcaa2.png">

Selecting a different global style, the editor updates while the preview doesn't:

<img width="1352" alt="Screenshot 2020-03-25 at 18 18 39" src="https://user-images.githubusercontent.com/1562646/77566574-26196580-6ec6-11ea-9f13-7043f528cbca.png">

**Comment by frontdevde** [comment]
Just to add to the above finding, I'm observing the same issue with Block Patterns' preview as well.

**Comment by frontdevde** [comment]
Working on a fix for https://github.com/Automattic/wp-calypso/issues/40467 which should, in theory, fix https://github.com/Automattic/wp-calypso/issues/40353 as well.

**Comment by frontdevde** [comment]
Fixed by https://github.com/Automattic/wp-calypso/pull/40489
